doc/design/parallel-node-config.md

```diff
@@ -0,0 +1,195 @@
+---
+title: Parallel SR-IOV configuration
+authors:
+  - e0ne
+reviewers:
+  - adrianchiris
+  - SchSeba
+creation-date: 18-07-2023
+last-updated: 18-07-2023
+---
+
+# Parallel SR-IOV configuration
+
+## Summary
+SR-IOV Network Operator configures NICs one by one. That means we’ll need to wait hours or even days to configure all
+NICs on large cluster deployments. With multi-NICs deployments Operator configures NICs one by one on each node which
+leads to a lot of unnecessary node drain calls during SR-IOV configuration.
+
+## Motivation
+Allow SR-IOV Network Operator to configure more than one node at the moment.
+
+### Use Cases
+
+### Goals
+* Number of drainable at the moment nodes should be 1 be default
+* Number of drainable at the moment nodes should be configured by pool
+* Nodes pool should be defined by node selector
+* A node could be included into several pools. In this case pool should be 
+  selected by it’s priority like it’s implemented for SriovNetworkNodePolicy
+
+
+### Non-Goals
+Parallel NICs configuration on the same node is out of scope of this proposal
+
+## Proposal
+
+Introduce nodes pool drain configuration to meet following requirements:
+* Number of drainable at the moment nodes should be 1 be default
+* Number of drainable at the moment nodes should be configured by pool
+* Nodes pool should be defined by node selector
+* A node could be included into several pools. In this case pool should be selected by it’s priority like it’s
+  implemented for `SriovNetworkNodePolicy`
+
+
+### Workflow Description
+A new Drain controller will be introduced to manage node drain and cordon procedures. That means we don't need to do
+drain and use `drain lock` in config daemon anymore. The overall drain process will be covered by the following states:
+
+```golang
+AnnoIdle            = "Idle"
+AnnoDrainRequired   = "Drain_Required"
+AnnoMcpPaused       = "Draining_MCP_Paused"
+AnnoDraining        = "Draining"
+AnnoDraining        = "Draining_Complete"
+```
+
+Drain controller will watch for node annotation changes, `SriovNetworkPoolConfig` and `SriovNetworkNodeState` changes:
+```golang
+// +kubebuilder:rbac:groups="",resources=nodes,verbs=get;list;watch;update;patch
+// +kubebuilder:rbac:groups="",resources=SriovNetworkPoolConfig,verbs=get;list;watch;update;patch
+// +kubebuilder:rbac:groups="",resources=SriovNetworkNodeState,verbs=get;list;watch;update;patch
+```
+
+Since new `SriovNetworkNodeState` will appear with `syncState=InProgess` controller will `Drain_Required` annotation.
+Another reconcile loop will check if current `Draining` nodes count is less than `maxUnavailable` and will
+mark it as `Draining` to start draining procedure. If current Draining nodes count equals to 
```

Comment:
nit: equals -> greater equal .

the greater equal is in case maxUnavailable changed for the node and we are now oversubscribed. (i think its ok to be in this state)